bids/grabbids/bids_layout.py

```diff
@@ -12,6 +14,53 @@ def __init__(self, path, config=None):
             config = os.path.join(root, 'config', 'bids.json')
         super(BIDSLayout, self).__init__(path, config)
 
+    def get_metadata(self, path):
+        sidecarJSON = path.replace(".nii.gz", ".json").replace(".nii", ".json")
+        pathComponents = os.path.split(sidecarJSON)
+        filenameComponents = pathComponents[-1].split("_")
+        ses = None
+        suffix = filenameComponents[-1]
+
+        sub = filenameComponents[0]
+        keyword_components = filenameComponents[1:-1]
+        if filenameComponents[1][:3] == "ses":
+            ses = filenameComponents[1]
+            keyword_components = filenameComponents[2:-1]
+
+        potentialJSONs = []
+        for k in range(len(keyword_components) + 1):
+            print(k)
+            for components in itertools.combinations(keyword_components, k):
+                print(components)
+                potentialJSONs.append(os.path.join(self.root,
+                                                   "_".join(components +
+                                                            (suffix,))))
+
+        for k in range(len(keyword_components) + 1):
+            for components in itertools.combinations(keyword_components, k):
+                potentialJSONs.append(os.path.join(self.root, sub,
+                                                   "_".join((sub,) +
```

Comment:
I don't quite understand... Maybe you can fix this in a separate PR? I have no love or passion for keeping lines within 79 characters...